### DISCUSSION TITLE
Branches in GitHub

### DISCUSSION THREAD

**Discussion by pakricard** [timschofield/webERP#267] [Development Discussion & Specification]
Hi:
I am getting confused about the branches https://github.com/timschofield/webERP. I can see:

- dashboard
- DBUpdate
- master
- new_gl
- release_4.15.1
- v5.0

The latest official release of webERP is 4.15.2, then:

- Master branch: Does it contain the v4.15.2 with bug fixes? Or does it contain the latest development too?
- Is there a branch that contains v4.15.2 with bug fixes for that version?
- When tacking the issues #260 or #261 (as example) belonging to v4.15.2 which branch to use to solve these issues? 

So far, I am committing to master, but becoming unsure if I am doing the right thing when fixing v4.15.2 stuff

**Comment by dalers** [comment]
Imho....

The master branch is what will _eventually_ be released as 5.0.0. Any development on the path to a v5 release can go in the master branch.

There have been too many major commits to the master branch for it to be considered a v4.15 maintance branch now (e.g. DomPDF, PHPSpreadsheet, new installer, new database update mechanism, etc. etc. etc.).

A v4.15 maintenance branch _could_ be created, likely starting with the 4.15.2 release commit, cherry pick commits from the master branch that are truly 4.15.2 bug fixes, and then continue with new maintenance commits (which could be cherry picked to the master branch if relevent to v5).

However....  I recall @pakricard you saying somewhere (I had thought in the #211 discussion of releases and the release process, but I can't find it there now), that process must be kept simple with a small project team, and supporting a v4.15 maintenance branch in addition to a v5 dev branch ("master") doesn't make things simpler (but still possible given enough motivation and time).

I suspect all the branches other than master could be deleted to cleanup the repository - and prevent confusion ;-)

**Comment by timschofield** [comment]
In my opinion, everything except master can go for now. Once v5 is
released we should have a v5 branch for any fixes, and master branch
would contain fixes and new functionality.

It's very useful for developers to keep multiple local branches - I
have 5 on my laptop currently, but GitHub should be kept simple.
Unless there are any strong opinions on this, I will remove the
others.

Thanks
Tim

On Mon, 2 Dec 2024 at 05:09, Dale Scott ***@***.***> wrote:
>
> Imho....
>
> The master branch is what will eventually be released as 5.0.0. Any development on the path to a v5 release can go in the master branch.
>
> There have been too many major commits to the master branch for it to be considered a v4.15 maintance branch (e.g. DomPDF, PHPSpreadsheet, new installer, new database update mechanism, etc. etc. etc.).
>
> A v4.15 maintenance branch could be created, starting with the 4.15.2 release commit and cherry pick commits to the master branch that are truly only 4.15.2 bug fixes. New v4.15 bug fix commits would be commited to the v4.15 maintenance branch and, if relevent to v5, cherry picked to the master branch.
>
> There was discussion of releases and the release process in issue #211 but I recall you saying somewhere else (although I can't find it now) that things have to be kept simple with a small project team, and supporting a v4.15 maintenance branch in addition to a v5 branch isn't making things simpler - but still possible given enough motivation and time.
>
> I suspect all the branches other than master could be deleted to cleanup the repository - and prevent confusion ;-)
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by dalers** [reply]
good plan @timschofield I agree with a) removing irrelevent branches, and b) your plan for a v5 maintenance branch when the time comes (perhaps even just a little bit early so it can be used for final release stabilization).

**Comment by timschofield** [comment]
Thanks Dale, I have removed these branches, they seemed to play no real
role, as they were all out of date.

Tim

On Mon, 2 Dec 2024 at 15:53, Dale Scott ***@***.***> wrote:

> good plan @timschofield <https://github.com/timschofield> I agree with a)
> removing irrelevent branches, and b) your plan for a v5 maintenance branch
> when the time comes (perhaps even just a little bit early so it can be used
> for final release stabilization).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/267#discussioncomment-11438267>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LAHO2HJE5V35CX5FJT2DR653AVCNFSM6AAAAABS2MBYGSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBTHAZDMNY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by pakricard** [comment]
Thanks, @timschofield.  The simpler, the better. 

Anyway, according to by SourceTree updated yesterday (when I created the discussion), the branch v5.0 wasn't merged into master and DBUpdate, dashboard, were merged into this v5.0.

![image](https://github.com/user-attachments/assets/0007179d-53c8-4512-96d4-b6d29af2a10c)


I find useful to create a new branch for every (relatively) large update or project involving more than 4–5 scripts or more than one day of work. It keeps the master branch always in a “working order”, as the new update or project can become more complex, or have side effects, etc are isolated in its branch. Once tested, I merge into master and delete the branch.

**Comment by pakricard** [comment]
Thinking about it...

https://github.com/timschofield/webERP/wiki/V5 states the milestones for v5.0.0 and v5.x. 
It would be great if we could segregate somehow the commits related to v5.x (maybe by subject) so whatever is on the master is the core we need to get up-and-running (hopefully with very few bugs) and, at a later time, we can focus on v5.x

**Comment by timschofield** [comment]
Hi Ricard, my thoughts at the moment was that everything being committed at
the moment would be in v5.0. At some point we must say that nothing new is
going in for 5.0, only bug fixes, and at that point we branch off a v5.0
branch and just apply bug fixes to that.

On Wed, 4 Dec 2024 at 10:06, pakricard ***@***.***> wrote:

> Thinking about it...
>
> https://github.com/timschofield/webERP/wiki/V5 states the milestones for
> v5.0.0 and v5.x.
> It would be great if we could segregate somehow the commits related to
> v5.x (maybe by subject) so whatever is on the master is the core we need to
> get up-and-running (hopefully with very few bugs) and, at a later time, we
> can focus on v5.x
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/267#discussioncomment-11459385>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LHGCFTULLKXPZHHHSD2D3H2XAVCNFSM6AAAAABS2MBYGSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBVHEZTQNI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by pakricard** [reply]
OK, @timschofield !

**Comment by timschofield** [comment]
In the end that script does no harm, and is there as a guide, not sure I
see the problem.

Tim

On Sat, 14 Dec 2024 at 18:23, Dale Scott ***@***.***> wrote:

> Carrying on from discussion topics #297
> <https://github.com/timschofield/webERP/discussions/297> and #309
> <https://github.com/timschofield/webERP/issues/309> and commits a6361b1
> <https://github.com/timschofield/webERP/commit/a6361b1787ac75b83e1f267a8f9c7420d8771740>,
> e27fcbc
> <https://github.com/timschofield/webERP/commit/e27fcbc63637ea7a5e972a4031a98e642cc43137>
> and f254756
> <https://github.com/timschofield/webERP/commit/f254756085d9d019da04bde9d852a10825a4b343>,
> and with the benefit of hindsight, would it have been better to commit the
> new LoggedInUsers.php script (and related work) to a feature branch instead
> of master for the community to review, discuss and possibly refine, without
> affecting v5 stabilization?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/267#discussioncomment-11568041>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LGGKF6NK7RYY2Y3SCT2FRZRJAVCNFSM6AAAAABS2MBYGSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJWHAYDIMI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/
